vaultlocker/shell.py

```diff
@@ -40,10 +40,17 @@ def _vault_client(config):
     """
     client = hvac.Client(
         url=config.get('vault', 'url'),
-        verify=config.get('vault', 'ca_bundle', fallback=True)
+        token=config.get('vault', 'token', fallback=None),
+        verify=config.get('vault', 'ca_bundle', fallback=True),
+        namespace=config.get('vault', 'namespace', fallback=None)
     )
-    client.auth_approle(config.get('vault', 'approle'),
-                        secret_id=config.get('vault', 'secret_id'))
+    if not config.get('vault', 'role_id', fallback=None) is None:
+        role_id=config.get('vault', 'role_id')
+    if not config.get('vault', 'approle', fallback=None) is None:
+        role_id=config.get('vault', 'approle')
```

Comment:
This might be a bit more elegant:
```
  role_id = config.get('vault', 'role_id', fallback=config.get('vault', 'approle'))
```

Reply:
I'll need to test that again, but it seems that the fallback=confi.get() will be resolved even if role_id is defined, meaning that it would expect both values in the config, thus me doing the above code logic (which is indeed not very pretty)

Reply:
So yeah, just tested again and this is what happens:
```
vaultlocker: No option 'approle' in section: 'vault'
```